src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -229,6 +230,11 @@ internal CookieContainer CookieContainer
             set
             {
                 CheckDisposedOrStarted();
+                if (value == null)
+                {
+                    throw new ArgumentNullException("CookieContainer", SR.net_http_invalid_cookiecontainer);
```

Comment:
> I understand. Desktop does throw an exception. For now, you should match HttpClientHandler/WinHttpHandler behavior for CoreFx which is to NOT throw an exception in the setter.
Ah. sorry @davidsh, when I started typing my answers, I did not find yours. I think I need to refresh this page immediately after every answer. My earlier note was irrelevant. Yes, I will do as you said earlier.